core/src/data/dataSource.cpp

```diff
@@ -38,19 +43,7 @@ void DataSource::setTileData(const TileID& _tileID, const std::shared_ptr<TileDa
         m_tileStore[_tileID] = _tileData;
 }
 
-void DataSource::setUrlTemplate(const std::string& _urlTemplate){
```

Comment:
Should we still keep a setUrlTemplate around? In case we want to modify url template of an existing dataSource!!!!

But that being said, there is nothing stopping to create a new dataSource with that new url template.